pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -157,13 +150,25 @@ func (r *Reconciler) reconcile(ctx context.Context, cronjob *v1alpha1.CronJobSou
 	}
 	cronjob.Status.MarkSink(sinkURI)
 
+	_, err = cron.ParseStandard(cronjob.Spec.Schedule)
+	if err != nil {
+		cronjob.Status.MarkInvalidSchedule("Invalid", "")
+		return fmt.Errorf("invalid schedule: %v", err)
+	}
+	cronjob.Status.MarkSchedule()
+
 	ra, err := r.createReceiveAdapter(ctx, cronjob, sinkURI)
 	if err != nil {
-		logging.FromContext(ctx).Error("Unable to create the receive adapter", zap.Error(err))
+		r.Logger.Error("Unable to create the receive adapter", zap.Error(err))
```

Comment:
`logging.FromContext(ctx)` is preferable. It includes things like the key that is currently being reconciled.